shared/test/TestValueEvaluators.cpp

```diff
@@ -0,0 +1,396 @@
+#include "Value.h"
+
+#include <algorithm>
+#include <atomic>
+#include <catch2/benchmark/catch_benchmark.hpp>
+#include <catch2/catch_get_random_seed.hpp>
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/matchers/catch_matchers_floating_point.hpp>
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <thread>
+#include <vector>
+
+TEST_CASE("GetPropertyValue tests", "[GetPropertyValue]") {
+    auto featureContext = std::make_shared<FeatureContext>(vtzero::GeomType::POINT, FeatureContext::mapType{}, 0);
+    auto featureStateManager = std::make_shared<Tiled2dMapVectorStateManager>();
+    EvaluationContext context = EvaluationContext(0, 0, featureContext, featureStateManager);
+
+    SECTION("Evaluate when key does not exist") {
+        auto evaluator1 = ValueEvaluator<std::string>(std::make_shared<GetPropertyValue>("key"));
+        auto result1 = evaluator1.getResult(EvaluationContext(0, 0, featureContext, featureStateManager), "fallback");
+        REQUIRE(result1 == "fallback");
+    }
+
+    SECTION("Evaluate when key exist") {
+        featureContext->propertiesMap = FeatureContext::mapType{{"key", "value"}};
+        auto staticValue = std::make_shared<GetPropertyValue>("key");
+        auto evaluator = ValueEvaluator<std::string>(staticValue);
+        auto result = evaluator.getResult(EvaluationContext(0, 0, featureContext, featureStateManager), "fallback");
```

Comment:
This test case uses ValueEvaluator, while all the tests below directly access `value.evaluate`.
Perhaps it makes sense to keep this to `value.evaluate` and add a more focused set of tests for ValueEvaluator with its caching logic.

Btw. perhaps this file should then be called "TestValueEvaluate.cpp"? :)